controller/backing_image_manager_controller.go

```diff
@@ -713,8 +726,22 @@ func (c *BackingImageManagerController) prepareBackingImageFiles(currentBIM *lon
 			continue
 		}
 
+		// Before syncing the backing image copy to this backing image manager,
+		// check if there are more than ReplenishPerNodeLimit number of backing image copies are being synced on this node.
+		canSync, err := c.CanSyncCopy(currentBIM, biName, currentBIM.Spec.NodeID, currentBIM.Spec.DiskUUID)
```

Comment:
nit:
I thought CanSyncCopy only checks if syncing copy is available, but I found it also delete or add entry to `c.inProgressReplenishingMap`?
I would suggest separating this functionality into two distinct functions: one for checking availability and another for modifying the map.

Reply:
The logic here is copied from replica: https://github.com/longhorn/longhorn-manager/blob/v1.6.2/controller/replica_controller.go#L436

the map is a in-memory map, so we iterate all the entries every time so it can rebuild if the controller or the pod crashes
we need to add and delete the map  so we can finally decide if it can sync the copy now